datafusion/functions-nested/src/string.rs

```diff
@@ -251,20 +254,21 @@ impl ScalarUDFImpl for StringToArray {
             Utf8 | LargeUtf8 => {
                 List(Arc::new(Field::new("item", arg_types[0].clone(), true)))
             }
+            Utf8View => List(Arc::new(Field::new("item", Utf8, true))),
```

Comment:
Should we use UTf8View for the element type?

Reply:
Possible though I am uncertain as to whether anything would benefit from it. Hardly any functions currently return Utf8View currently.

Reply:
unclear to me -- if we're creating new string values, would there be benefit of using Utf8View?

Reply:
If a function called after one of these functions were called that would actually materially benefit from utf8view vs utf8 would it really matter. The main difference in actual code would be requiring a different StringBuilder in the `string_to_array_impl` function.

Reply:
yes, indeed. is contrary also possible?
also, isn't this the downstream function's problem, not here?

ideally in https://github.com/apache/datafusion/issues/12622, utf8view would be run-time variant of utf8, same logical type, so next function could choose on its own

Reply:
> yes, indeed. is contrary also possible? also, isn't this the downstream function's problem, not here?

True. However since few other functions return utf8vew (currently) it would generally mean that at some point any system that chains functions (fairly normal dataframe usage) the output is going to be utf8 anyways. I'm not against adding it here at all, just being realistic as to what conversions are going to happen in a lot of cases anyways.

We may want to create an epic to update all the functions such that the output matches the input type (if input and output are string types) similar to what we did for utf8view for input.

> ideally in #12622, utf8view would be run-time variant of utf8, same logical type, so next function could choose on its own

Yes, that would be the case for input.